R/post-body.R

```diff
@@ -24,3 +30,27 @@ parseBody <- function(body){
   }
   ret
 }
+
+#' @importFrom utils URLdecode
+#' @noRd
+parseMultipart <- function(body){
+  # This is not as generic as perhaps it should be, but should be relatively
+  # easy to extend if more binary formats are required
+  # Is there data in the request?
+  if (is.null(body) || length(body) == 0 || body == "") {
+    return(list())
+  }
+  # This is something odd with webutils::parse_multipart. For some reason it
+  # does not work when the boundary is set to the content type as per the
+  # webutils docs.
+  boundary <- "-----------------"
+  parsed_binary <- webutils::parse_multipart(body, boundary)
+  file_name <- parsed_binary[[1]][[1]]
+  file_data <- parsed_binary[[1]][[2]]
+
+  tmpfile <- tempfile()
+  writeBin(file_data, tmpfile)
```

Comment:
For my purpose I would prefer to be able to land the files on disk. If we use textConnections, the data would not be persistent at all, and not be available after a restart of the service. 

I can imagine use cases where persistence would not be an issue. Maybe this is something to make a separate ticket of, to add as an option later.